### PR TITLE
add dbvar store property library

### PR DESCRIPTION
**Pull request by paolino**
- [x] factor out the store property test in DBVar lib

### Issue Number

ADP-1784

### CODE REVIEW

**Comment by paolino** [comment]
bors try

File: lib/dbvar/src/Test/DBVar.hs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+module Test.DBVar
+    ( genUpdates,prop_StoreUpdates, GenDelta, Updates (..)) where
```
Comment (HeinrichApfelmus) [inline lib/dbvar/src/Test/DBVar.hs]:
```suggestion
module Test.DBVar
    ( genUpdates
    , prop_StoreUpdates
    , GenDelta
    , Updates (..)
    ) where
```

File: lib/dbvar/src/Test/DBVar.hs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    -- check whether the last value is correct
+    case ea of
+        Left err -> impureThrow err
+        Right a  -> assert $ a == head as
+
+
+{-------------------------------------------------------------------------------
+    Properties
+-------------------------------------------------------------------------------}
```
Comment (HeinrichApfelmus) [inline lib/dbvar/src/Test/DBVar.hs]:
```suggestion
```
Leftover from rebase?

**Review by HeinrichApfelmus — APPROVED** [review]
Did I mention that it's a pleasure to review? 😄 Tentatively approved, pending two changes.

File: lib/dbvar/src/Test/DBVar.hs (hunk 2 above)
Comment (paolino) [reply lib/dbvar/src/Test/DBVar.hs]: Well, leftover, fullstop. :-/

**Pull request merged**